libs/model/src/community/UpdateCommunity.command.ts

```diff
@@ -0,0 +1,160 @@
+import { InvalidInput, type Command } from '@hicommonwealth/core';
+import * as schemas from '@hicommonwealth/schemas';
+import { ALL_COMMUNITIES, ChainBase } from '@hicommonwealth/shared';
+import { models } from '../database';
+import { AuthContext, isAuthorized } from '../middleware';
+import { mustExist } from '../middleware/guards';
+import { checkSnapshotObjectExists, commonProtocol } from '../services';
+
+export const UpdateCommunityErrors = {
+  ReservedId: 'The id is reserved and cannot be used',
+  CantChangeCustomDomain: 'Custom domain change not permitted',
+  CustomDomainIsTaken: 'Custom domain is taken by another community',
+  CantChangeNetwork: 'Cannot change community network',
+  NotAdmin: 'Not an admin',
+  InvalidCustomDomain: 'Custom domain may not include "commonwealth"',
+  SnapshotOnlyOnEthereum:
+    'Snapshot data may only be added to chains with Ethereum base',
+  InvalidDefaultPage: 'Default page does not exist',
+  InvalidTransactionHash: 'Valid transaction hash required to verify namespace',
+  SnapshotNotFound: 'Snapshot not found',
+};
+
+export function UpdateCommunity(): Command<
+  typeof schemas.UpdateCommunity,
+  AuthContext
+> {
+  return {
+    ...schemas.UpdateCommunity,
+    auth: [isAuthorized({ roles: ['admin'] })],
+    body: async ({ actor, payload }) => {
+      const {
+        id,
+        network,
+        custom_domain,
+        snapshot,
+        name,
+        description,
+        default_symbol,
+        icon_url,
+        active,
+        type,
+        stages_enabled,
+        has_homepage,
+        chain_node_id,
+        directory_page_enabled,
+        directory_page_chain_node_id,
+        default_summary_view,
+        default_page,
+        social_links,
+        hide_projects,
+        custom_stages,
+        namespace,
+        transactionHash,
+      } = payload;
+
+      if (id === ALL_COMMUNITIES)
+        throw new InvalidInput(UpdateCommunityErrors.ReservedId);
+      if (network)
+        throw new InvalidInput(UpdateCommunityErrors.CantChangeNetwork);
+
+      const community = await models.Community.findOne({
+        where: { id },
+        include: [
+          {
+            model: models.ChainNode,
+            attributes: ['url', 'eth_chain_id', 'cosmos_chain_id'],
+          },
+        ],
+      });
+      mustExist('Community', community); // if authorized as admin, community is always found
+
+      // Only permit site admins to update custom domain field on communities, as it requires
+      // external configuration (via heroku + whitelists).
+      // Currently does not permit unsetting the custom domain; must be done manually.
+      if (custom_domain && community.custom_domain !== custom_domain) {
```

Comment:
Shouldn't this be entirely disabled i.e. `custom_domain` should not be in the payload? @kurtisassad

Reply:
Yeah we can remove this because the custom domain stuff is in a different route now.

Reply:
Confirmed with Kurtis that this should be removed.

Reply:
Removed 👍